lib/getReleases.js

```diff
@@ -1,26 +1,95 @@
 
 // @ts-check
-const fetch = require('node-fetch');
+const fetch = require('node-fetch').default;
 const { lt } = require('semver');
 
-class GitHubAPIError extends Error {
-  /**
-   * @param {string} message
-   * @param {any} [cause]
-   */
-  constructor(message, cause) {
-    super(message);
-    this.cause = cause;
-    this.name = 'GitHub GraphQL Error';
-  }
-}
-
-exports.findLatestVSIXRelease = async function (/** @type {string} */ repository, /** @type {string} */ version) {
+/**
+ * 
+ * @param {string} repository 
+ * @param {string} version 
+ * @param {string | undefined} [msVersion]
+ * @returns {Promise<string | undefined>}
+ */
+exports.resolveFromRelease = async function (repository, version, msVersion) {
   const [owner, repo] = new URL(repository).pathname.slice(1).split("/");
   console.log(`Fetching ${owner}/${repo} to check for updates...`);
   const token = process.env.GITHUB_TOKEN;
   if (!token) {
-    throw new Error("GitHub personal access token is missing");
+    throw new Error("GITHUB_TOKEN env var is not set");
+  }
+
+  if (msVersion) {
+    // TODO(ak) replace with proper statically typed npm package
+    const response = await fetch("https://api.github.com/graphql", {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json', 'Authorization': `Bearer ${token}` },
+      body: JSON.stringify({
+        query: `query GetVSIXFromRelease {
+          repository(name: "${repo}", owner: "${owner}") {
+            release(tagName: "${msVersion}") {
+              id
+              tagName
+              releaseAssets(first: 20) {
+                edges {
+                  node {
+                    downloadUrl
+                  }
+                }
+              }
+            }
+          }
+        }` }),
+    });
+    if (!response.ok) {
+      throw new Error(response.statusText);
+    }
+
+    const result = await response.json();
+    if (!result.data && result.errors) {
+      throw new Error(JSON.stringify(result.errors));
+    }
```

Comment:
**suggestion (non-blocking):** Why not use `try catch` instead of checking errors in the if condition?

Reply:
Well if we wanted to try catch, we would need to raise the exception somewhere, conditionally, this cannot be done in a `try catch` block.